app/actions/space_delete.rb

```diff
@@ -82,7 +82,7 @@ def delete_apps(space_model)
     def delete_service_brokers(space_model)
       broker_remover = service_broker_remover(@services_event_repository)
       private_service_brokers = space_model.service_brokers
-      deletable_brokers = private_service_brokers.reject { |broker| broker.service_plans.map(&:service_instances).flatten.any? }
+      deletable_brokers = private_service_brokers.reject { |broker| broker.service_plans.map { |plan| plan.service_instances_dataset.any? }.flatten.any? }
```

Comment:
I actually had a look at this a few days ago and came up with the following:
```
deletable_brokers = private_service_brokers.reject do |broker|
  # Brokers with existing service instances are not deletable and thus rejected from the array.
  ServiceInstance.
    join(:service_plans, id: :service_instances__service_plan_id).
    join(:services, id: :service_plans__service_id).
    where(services__service_broker_id: broker.id).
    any?
end
```

Its (a) reducing the number of queries (m instead of m * n) and (b) reducing the amount of data fetched from the database (m boolean values instead of m service plans * n service instances).